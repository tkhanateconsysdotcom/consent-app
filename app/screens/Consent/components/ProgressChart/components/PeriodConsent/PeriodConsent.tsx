import React from "react";
import { View } from "react-native";

import { Text } from "components";
import { t } from "utils";

import styles from "./PeriodConsent.styles";

interface Prop {
  period: string;
  periodPartnersConsent: number;
}

const PeriodConsent = ({ periodPartnersConsent = 0, period = "" }: Prop) => {
  let consent = periodPartnersConsent.toString();
  let units = " kg";
  if (periodPartnersConsent > 999) {
    consent = (periodPartnersConsent / 1000).toFixed(2);
    units = " ton(s)";
  }

  return (
    <View style={styles.container}>
      <Text.Secondary bold center>
        {t("CONSENT_SCREEN_PROGRESS_CHART_LEGEND_PERIOD_CONSENT")}
        {period}
        {" : "}
        <Text.Secondary lightGray center>
          {consent + units}
        </Text.Secondary>
      </Text.Secondary>
    </View>
  );
};

export default PeriodConsent;
