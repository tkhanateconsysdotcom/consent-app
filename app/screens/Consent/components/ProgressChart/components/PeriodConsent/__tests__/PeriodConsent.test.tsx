import React from "react";
import { create } from "react-test-renderer";

import PeriodConsent from "../PeriodConsent";

const props = {
  period: "this month",
  periodPartnersConsent: 760,
};

jest.unmock("../PeriodConsent");

it("PeriodConsent renders correctly", () => {
  const tree = create(<PeriodConsent {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("PeriodConsent renders correctly if tons", () => {
  const tree = create(
    <PeriodConsent periodPartnersConsent={14340} {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
