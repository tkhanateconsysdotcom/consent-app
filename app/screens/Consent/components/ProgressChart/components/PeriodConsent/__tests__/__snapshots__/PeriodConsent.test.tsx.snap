// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodConsent renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 16,
      "paddingTop": 12,
    }
  }
>
  <Text.Secondary
    bold={true}
    center={true}
  >
    Consent for 
    this month
     : 
    <Text.Secondary
      center={true}
      lightGray={true}
    >
      760 kg
    </Text.Secondary>
  </Text.Secondary>
</View>
`;

exports[`PeriodConsent renders correctly if tons 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 16,
      "paddingTop": 12,
    }
  }
>
  <Text.Secondary
    bold={true}
    center={true}
  >
    Consent for 
    this month
     : 
    <Text.Secondary
      center={true}
      lightGray={true}
    >
      760 kg
    </Text.Secondary>
  </Text.Secondary>
</View>
`;
